Calendar

n1313 opened the pull request:
See also https://github.com/FreeFeed/freefeed-server/pull/621

n1313 committed:
Add calendar

davidmz commented on src/components/calendar/calendar-date.jsx:
```diff
@@ -0,0 +1,81 @@
+  return { ...postActions(dispatch) };
+}
+
+export default connect(mapStateToProps, mapActionsToDispatch)(CalendarYear);
```
А давай лучше через useSelecter/useDispatch? Раз уж всё равно функциональный компонент у нас.

davidmz commented on src/components/calendar/calendar-year.jsx:
```diff
@@ -0,0 +1,144 @@
+          {nextYear <= thisYear ? (
+            <Link to={`/${userName}/calendar/${nextYear}`}>{nextYear} &rarr;</Link>
+          ) : (
+            <span>{nextYear} &rarr;</span>
```
А вот это зачем? Меня удивил 2024-й год на странице, пусть и неактивный.

n1313 replied:
Для того, чтобы с центровкой "2023" не возиться :)

n1313 committed:
Add weekday display

n1313 committed:
Add calendar utilities

n1313 committed:
Add calendar month view

n1313 commented:
Added month view, weekday names, refactored some utils.


<img width="744" alt="Screenshot 2023-04-24 at 19 47 06" src="https://user-images.githubusercontent.com/632081/234162285-f0f97f58-f167-4176-b9d2-4edb93d855a9.png">
<img width="236" alt="Screenshot 2023-04-24 at 19 47 16" src="https://user-images.githubusercontent.com/632081/234162291-4805c678-5305-4147-af7f-21af2ae8c63e.png">

n1313 committed:
Add single month days grid view

n1313 committed:
Improve previous/next date navigation

n1313 commented:
<img width="719" alt="Screenshot 2023-04-30 at 00 07 33" src="https://user-images.githubusercontent.com/632081/235362984-2db6ae58-dc5c-43d9-a69c-14a4dc23bf9f.png">
<img width="723" alt="Screenshot 2023-04-30 at 00 08 35" src="https://user-images.githubusercontent.com/632081/235362992-e9fdcb55-b017-4329-b6d3-fc7cd429981d.png">

n1313 committed:
Use date-fns to format dates instead of doing it manually like a barbarian

n1313 committed:
Add a link to calendar view to "More" menu
